cli/src/cmd/forge/test.rs

```diff
@@ -390,24 +298,123 @@ fn short_test_result(name: &str, result: &forge::TestResult) {
+    let verbosity = evm_opts.verbosity;
+    if args.gas_report && evm_opts.verbosity < 3 {
+        evm_opts.verbosity = 3;
+    }
+
+    // Prepare the test builder
+    let evm_spec = crate::utils::evm_spec(&config.evm_version);
+    let mut runner = MultiContractRunnerBuilder::default()
+        .fuzzer(fuzzer)
+        .initial_balance(evm_opts.initial_balance)
+        .evm_spec(evm_spec)
+        .sender(evm_opts.sender)
+        .with_fork(utils::get_fork(&evm_opts, &config.rpc_storage_caching))
+        .build(project.paths.root, output, evm_opts)?;
+
+    if args.debug.is_some() {
+        args.filter.test_pattern = args.debug;
+        match runner.count_filtered_tests(&args.filter) {
+                1 => {
+                    // Run the test
+                    let results = runner.test(&args.filter, None, true)?;
+
+                    // Get the result of the single test
+                    let (id, sig, test_kind, counterexample) = results.iter().map(|(id, SuiteResult{ test_results, .. })| {
+                        let (sig, result) = test_results.iter().next().unwrap();
+
+                        (id.clone(), sig.clone(), result.kind.clone(), result.counterexample.clone())
+                    }).next().unwrap();
+
+                    // Build debugger args if this is a fuzz test
+                    let sig = match test_kind {
+                        TestKind::Fuzz(cases) => {
+                            if let Some(counterexample) = counterexample {
+                                counterexample.calldata.to_string()
+                            } else {
+                                cases.cases().first().expect("no fuzz cases run").calldata.to_string()
+                            }
+                        },
+                        _ => sig,
+                    };
+
+                    // Run the debugger
+                    let debugger = RunArgs {
+                        path: PathBuf::from(runner.source_paths.get(&id).unwrap()),
+                        target_contract: Some(utils::get_contract_name(&id).to_string()),
+                        sig,
+                        args: Vec::new(),
+                        debug: true,
+                        opts: args.opts,
+                        evm_opts: args.evm_opts,
+                    };
+                    debugger.run()?;
+
+                    Ok(TestOutcome::new(results, args.allow_failure))
+                }
+                n =>
+                    Err(
+                    eyre::eyre!("{} tests matched your criteria, but exactly 1 test must match in order to run the debugger.\n
+                        \n
+                        Use --match-contract and --match-path to further limit the search.", n))
+            }
+    } else {
+        let TestArgs { filter, .. } = args;
+        test(
+            runner,
+            verbosity,
+            filter,
+            args.json,
+            args.allow_failure,
+            include_fuzz_tests,
```

Comment:
that's the only thing that changed after moving this function, right?

Reply:
yup!